.github/actions/docker-dump-containers-logs/action.yml

```diff
@@ -0,0 +1,21 @@
+name: "Docker dump containers logs"
+description: "Persists Docker containers logs"
+runs:
+  using: composite
+  steps:
+    - name: "Dump containers logs and archive them"
+      shell: bash
+      run: |
+        # Get containers state
+        docker ps -a
+        # Dump all containers logs to files
+        docker ps -a --format '{{.ID}} {{.Names}}' | while read -r id name ; do
+          docker logs "${id}" --tail 5000 > "${name}.log"
+        done
+        # Create archive with log files
+        tar -zcf logs.tar.gz *.log
+    - name: "Upload archive containing all *.log files"
+      uses: actions/upload-artifact@v3
+      with:
+        path: logs.tar.gz
+        name: containers-logs.tar.gz
```

Comment:
In retrospective, if we wish to use this action in several jobs/steps within the same workflow, giving it a static name will make it so that the last artifact wins and overrides the previous one as far as I understand.
Shouldn't we add some uniqueness to this artifact's name so that it can be used more and with less preoccupations?

@gionn not sure if you have any suggestions regarding the name uniqueness, some that come to mind:
- time-based _(e.g.: including the evaluated `$(date +%s)` in the artifact name)_
- random-based _(.e.g.: including a UUID in the artifact name)_
- input-based _(governed by an input in this action, so that the users can easily identify which artifact is linked to which step/job as the final artifact name would be something like `${INPUT_IDENTIFIER}-containers-logs.tar.gz`)_

Ideally it could also be a mix and match of these solutions, where if the user does not specify any input than a pseudo-random unique name will be used _(e.g. via the time-based approach)_, whereas if the user does specify the input then that will be preferred. So we keep this super easy to use as there is no required configuration, but at the same time we make it easier to use efficiently in bigger workflows.

Reply:
Good point Domenico! I'm in favor of mixed solution - to allow users to specify the output archive name entirely, and if not specified then use the pseudo random one. I would also consider evaluating `$(date +%Y%m%d%H%M%S)` as it's more meaningful then epoch seconds.

Reply:
`${ github.job }` which reference the job id and `${ github.run_attempt }` which contains the retry number (starting from 1)?

Reply:
I'm testing the solution with job id and retry number, but I'm afraid it won't be applicable in case of usage in multiple steps within a job. Adding timestamp might be necessary.